src/java.base/share/classes/java/lang/constant/MethodTypeDescImpl.java

```diff
@@ -71,8 +76,20 @@ final class MethodTypeDescImpl implements MethodTypeDesc {
     static MethodTypeDescImpl ofDescriptor(String descriptor) {
         requireNonNull(descriptor);
         List<String> types = ConstantUtils.parseMethodDescriptor(descriptor);
-        ClassDesc[] paramTypes = types.stream().skip(1).map(ClassDesc::ofDescriptor).toArray(ClassDesc[]::new);
-        return new MethodTypeDescImpl(ClassDesc.ofDescriptor(types.get(0)), paramTypes);
+        ClassDesc[] paramTypes;
+        int paramCount = types.size() - 1;
+        if (paramCount > 0) {
+            paramTypes = new ClassDesc[paramCount];
+            for (int i = 0; i < paramCount; i++) {
+                paramTypes[i] = validateParameter(ClassDesc.ofDescriptor(types.get(i + 1)));
```

Comment:
It seems useful to have a static factory method to take a trusted copy of parameter types and it will validate the parameters before constructing `MethodTypeDescImpl` instance.   Parameter validation in one single place.    Several methods doing the validation can simply call this factory method and the code would be cleaner.

`insertParameterTypes` can call it as well and overhead of re-validating existing parameter types isn't a big issue.

Reply:
Should I just keep track of the slot count of MethodTypeDesc instead? Having a slot count implicitly requires validating the parameters, and the slot count can be used to eagerly reject `resolveConstantDesc` on MTD with over 255 slots.